refactor: get_headers conformance tests

Mirko-von-Leipzig opened the pull request:
Last of the conformance test refactoring.

niklaslong commented on src/tests/conformance/query/get_header.rs:
```diff
@@ -1,139 +1,344 @@
-    let (_, headers) = synthetic_node.recv_message_timeout(TIMEOUT).await.unwrap();
-    let headers = assert_matches!(headers, Message::Headers(headers) => headers);
-    assert_eq!(headers.headers, expected[2..], "test for forked Headers");
+    let response = if reply.is_empty() {
```
nit/suggestion: `match` on `reply.len()`

niklaslong reviewed (approved):
Nice! 🙌 

Would `basic_query.rs` fit in this new `query` dir?

Mirko-von-Leipzig commented:
> 
> 
> Nice! 🙌
> 
> Would `basic_query.rs` fit in this new `query` dir?

I've moved the file in, and fixed the `match len` nit (x2).

I also renamed `get_header.rs` to be plural - this matches the message struct etc.
